### PR TITLE
chore: fix release build scripts

### PR DESCRIPTION
**Pull request by lesam**
Note: this commit fixes linux/amd64 and darwin/amd64 builds. We are still
working on fixes for linux/arm64 and windows/amd64.

Partial fix of https://github.com/influxdata/influxdb/issues/20854

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: releng/raw-binaries/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,54 @@
 ARG GO_VERSION
 FROM golang:${GO_VERSION}
 
+# Install system dependencies + cross build system dependencies
 RUN apt-get update && apt-get install -y --no-install-recommends \
-  jq \
+    asciidoc \
```
Comment (danxmoran) [inline releng/raw-binaries/Dockerfile:6]: Curious to know what `asciidoc` is being used for

File: releng/raw-binaries/fs/usr/local/bin/influxdb_raw_binaries.bash [hunk 2]
```diff
@@ -15,29 +17,62 @@ To build with race detection enabled, pass the -r flag.
...
+echo "Extracting influxdb tarball"
+# Extract tarball into WORKSPACE.
 
-SHA=$(jq -r .sha < "$GOPATH/src/github.com/influxdata/influxdb/.metadata.json")
+tar -vxz -C ${WORKSPACE} -f /influxdb-src.tar.gz
 
+SHA=$(jq -r .sha < "${WORKSPACE}/influxdb/.metadata.json")
+
+OUTDIR=$(mktemp -d)
+(
```
Comment (danxmoran) [inline releng/raw-binaries/fs/usr/local/bin/influxdb_raw_binaries.bash:52]: Why the subshell here?

File: releng/raw-binaries/fs/usr/local/bin/influxdb_raw_binaries.bash [hunk 3]
```diff
@@ -15,29 +17,62 @@ To build with race detection enabled, pass the -r flag.
...
+	cd ${WORKSPACE}/influxdb
+
+	BINARY_PACKAGES="
+		github.com/influxdata/influxdb/cmd/influxd
+		github.com/influxdata/influxdb/cmd/influx
+		github.com/influxdata/influxdb/cmd/influx_inspect
+		github.com/influxdata/influxdb/cmd/influx_tsm"
+
+	for cmd in $BINARY_PACKAGES; do
+		export CGO_ENABLED=1
```
Comment (danxmoran) [inline releng/raw-binaries/fs/usr/local/bin/influxdb_raw_binaries.bash:62]: Nit: could move this out of the loop

File: releng/raw-binaries/Dockerfile (hunk 1 above)
Comment (lesam) [reply releng/raw-binaries/Dockerfile:6]: asciidoc is used for compiling `man` pages

File: releng/raw-binaries/fs/usr/local/bin/influxdb_raw_binaries.bash (hunk 2 above)
Comment (lesam) [reply releng/raw-binaries/fs/usr/local/bin/influxdb_raw_binaries.bash:52]: the `cd` happens in the subshell without affecting the working directory outside the sub shell

**Review by danxmoran — APPROVED** [review]

**Pull request merged**